Client/Coordinators/Router/DefaultRouter.swift

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+
+class DefaultRouter: NSObject, Router {
+    private var completions: [UIViewController: () -> Void]
```

Comment:
Is it possible to extend the `Router` and implement most of the methods instead of having a `DefaultRouter` instance?

Reply:
Protocol extensions should only really be used if you have (or are likely to have) multiple conformers to the protocol that all have the same implementation of the method. How is it possible to avoid having an instance?

Reply:
I'm not sure we can avoid having a `DefaultRouter`. We have to instantiate a Router, and it should always be this default one (i.e. there shouldn't be other Router types).

Reply:
"We have to instantiate a Router" yes in the end when using the router we will have to create an instance so that means `DefaultRouter` stays.

 "multiple conformers to the protocol that all have the same implementation of the method" for me a Router is precisely that. It's wrapping a navigation controller and mimics the presentation. I find it hard to believe other routers can do something different. Similar to classes, methods implemented in protocol extensions can also be overwritten.

I don't have a strong opinion on that, so I don't want to sound like the change is necessary. I guess in time when we will have multiple Routers and see that the code is repeating we can optimize it.

Reply:
We'll just instantiate `let router = DefaultRouter(navigationController: navigationController)` to get routers. So there's not supposed to be much repetition as will always instantiate the default router.